PWGHF/HFC/Tasks/taskCorrelationHfeHadrons.cxx

```diff
@@ -0,0 +1,284 @@
+			Deta = etaEle - etaHad;
+			if(SparseNo >= 0) hEHcorrArray[SparseNo]->Fill(ptEle, ptHad, Dphi, Deta);
+		}
+	}
+
+	Filter CollisionFilter = nabs(aod::collision::posZ) < 10.f && aod::collision::numContrib > (uint16_t) 1;
+	using aodCollisions = soa::Filtered<soa::Join<aod::Collisions,aod::Mults, aod::EvSels>>::iterator;
+	void process(aodCollisions const& collision , aod::EMCALClusters const& mAnalysisClusters, o2::aod::EMCALMatchedTracks const& MatchedTracks, GTrks  const& tracks)                      
+	{ 
+		if (!(Isrun3 ? collision.sel8() : (collision.sel7() && collision.alias_bit(kINT7)))) return;
+
+		registry.fill(HIST("hNevents"), 1);
+		registry.fill(HIST("zvertex"), collision.posZ());
+
+		/////////////////////////////////
+		// cluster info before match  ///
+		///////////////////////////////
+		if(fclusterinfo){
+			for (const auto& clusterbf:mAnalysisClusters)
+			{
+				registry.fill(HIST("fSprsClusterInfoBe"),clusterbf.energy(),clusterbf.eta(),clusterbf.phi(),clusterbf.nCells(),clusterbf.time());
+			}
+		}
+		int PassEMCal;
+		Double_t phiTrack = -999;
+		Double_t etaTrack = -999;
+		Double_t pTrack =  -999;
+		Double_t ptTrack =  -999;
+		Double_t dcaxyTrack=  -999;
+		Double_t dcazTrack =  -999;
+		Double_t tpcnsigmaTrack =  -999;
+
+		for (auto& Track : tracks) {
+
+			phiTrack = Track.phi();
+			etaTrack = Track.eta();
+			pTrack =  Track.p();
+			ptTrack =  Track.pt();
+			dcaxyTrack=   Track.dcaXY();
+			dcazTrack =  Track.dcaZ();
+			tpcnsigmaTrack =  Track.tpcNSigmaEl();
+
+			if(!Track.isGlobalTrackWoDCA())  continue;
+			PassEMCal = 0;
+			if((phiTrack > phiTrackMinEM && phiTrack  < phiTrackMaxEM) &&  (etaTrack > etaTrackMin && etaTrack < etaTrackMax))   PassEMCal = 1; //EMcal acceptance passed
+			if ((phiTrack  > phiTrackMinDc && phiTrack  < phiTrackMaxDC) && ((etaTrack > etaTrackMinDcright && etaTrack < etaTrackMaxDcright) || (etaTrack > etaTrackMinDcleft && etaTrack < etaTrackMaxDcleft))) PassEMCal = 2; //Dcal acceptance passed
+
+			registry.fill(HIST("fSprsdEdxnSigmaPt"), Track.tpcSignal(),tpcnsigmaTrack ,pTrack , ptTrack ,etaTrack ,phiTrack , dcaxyTrack , dcazTrack , PassEMCal); //track infor after filter bit
+
+			// Apply Track cut
+			if(ptTrack < ptTrackMin)  continue;  
+			if(abs(dcaxyTrack) > dcaXYTrackMax || abs(dcazTrack)> dcaZTrackMax) continue;
+
+			auto tracksofcluster = MatchedTracks.sliceBy(perClusterMatchedTracks, Track.globalIndex());
+			Double_t phiMatchTrack = -999;
+			Double_t etaMatchTrack = -999;
+			Double_t pMatchTrack =  -999;
+			Double_t ptMatchTrack =  -999;
+			Double_t tpcnsigmaMatchTrack =  -999;
+			Double_t phiMatchCluster = -999;
+			Double_t etaMatchCluster = -999;
+			Double_t EMatchCluster =  -999;
+			Double_t m02MatchCluster = -999;
+			Double_t m20MatchCluster =  -999;
+			Double_t timeMatchCluster   =  -999;
+			for(const auto& emtrack : tracksofcluster){
+
+				if(Track.globalIndex()!=emtrack.trackId()) continue;
```

Comment:
Why is this check needed? Doesn't `sliceBy` take care of that?

Reply:
No, for each global track, we get 5 EMCal matched tracks, that are vicinity to global track matched cluster